src/components/sbb-header/sbb-header.scss

```diff
@@ -39,26 +33,39 @@
 }
 
 :host([hide-on-scroll][data-fixed][data-visible]) {
-  --sbb-header-inset-block-start: 0;
-
   // Show transition
   --sbb-header-transition-timing: cubic-bezier(0, 0, 0.2, 1);
 }
 
 .sbb-header {
   position: var(--sbb-header-position);
   inset: 0 0 auto;
-  inset-block-start: var(--sbb-header-inset-block-start);
   background: var(--sbb-color-white-default);
   z-index: 10;
-  transition-property: var(--sbb-header-transition-property);
-  transition-duration: var(--sbb-header-transition-duration);
-  transition-timing-function: var(--sbb-header-transition-timing);
+  transition: {
+    property: var(--sbb-header-transition-property);
+    duration: var(--sbb-header-transition-duration);
+    timing-function: var(--sbb-header-transition-timing);
+  }
+  animation: {
+    duration: var(--sbb-header-transition-duration);
+    timing-function: var(--sbb-header-transition-timing);
+  }
 
   :host([data-shadow]) & {
     @include sbb.shadow-level-9-soft;
   }
 
+  :host([hide-on-scroll][data-fixed]) & {
+    animation-name: hide;
+    transform: translate3d(0, -100%, 0);
+  }
+
+  :host([hide-on-scroll][data-fixed][data-visible]) & {
+    animation-name: show;
+    transform: translate3d(0, 0, 0);
+  }
```

Comment:
What about introducing css vars and toggle the vars on host level where all other vars are toggled?